doc/add_board_support.md

```diff
@@ -1,30 +1,92 @@
-1. replace all base board names in string literals with the new board name
-1. add the new board to the features at src/riot-rs-boards/Cargo.toml, e.g.,
+The more similar a board is to one that is already supported, the easier.
+It is usually best to copy & adapt an existing one.
 
-        new_board = { optional=true, path="new_board" }
+- ensure [embassy architecture](#adding-an-embassy-architecture) is supported
+- in laze-project.yml:
+  - add "context" for the mcu (if it doesn't exist, yet)
+    - parent: the closest embassy arch
+      - selects: a [rustc-target](#adding-a-rustc-target-module) module
+      - if the architecture does not have a dedicated SWI, choose one now
+        and set CONFIG_SWI
+      - if the mcu is supported by probe-rs, add PROBE_RS_CHIP and PROBE_RS_PROTOCOL
+  - add "builder" for the actual board
 
-1. add an entry in src/riot-rs-boards/src/lib.rs. replace underlines in the
-   `pub use` but not in the feature name, e.g., if the board and feature
-   `nrf52840-mdk` -> `pub use nrf52840_mdk as board;`.
+Whether adding an intermediate context or just a builder depends if the MCU
+specific code can be re-used.
 
-1. in laze-project.yml, copy the base board builder entry (and fix the name)
+Example for the st-nucleo-f401re:
 
-1. try it: `laze -Cexamples/bottles build -b <new-board>`
+```yaml
+contexts:
+# ...
+- name: stm32f401retx
+  parent: stm32
+  selects:
+    - thumbv7em-none-eabi # actually eabihf, but riot-rs doesn't support hard float yet
+  env:
+  PROBE_RS_CHIP: STM32F401RETx
+  PROBE_RS_PROTOCOL: swd
+  RUSTFLAGS:
+    - --cfg context=\"stm32f401retx\"
+  CARGO_ENV:
+    - CONFIG_SWI=USART2
 
-Assuming the board can be flashed with openocd/jlink, the flash task should work
-as is. Otherwise, the flashing method needs to be added/fixed.
+builders:
+# ...
+- name: st-nucleo-f401re
+  parent: stm32f401retx
+
+```
+
+- src/riot-rs-boards/$BOARD: add crate that matches board name
+  - this crate should inject the board specific dependencies to the arch crates.
+- src/riot-rs-boards:
+  - Cargo.toml: add feature that matches board name
+  - src/lib.rs: add board to dispatch
+
+## Adding a rustc target module
+
+Each actual rustc target needs its own module in laze-project.yml.
+If the the device that's being added isn't listed yet, you'll need to take care
+of that.
+
+Example:
+
+```yaml
+modules:
+# ...
+  - name: thumbv6m-none-eabi
+    depends:
+      - cortex-m
+    env:
+      global:
+        RUSTC_TARGET: thumbv6m-none-eabi
+        CARGO_TARGET_PREFIX: CARGO_TARGET_THUMBV6M_NONE_EABI
+        RUSTFLAGS:
+          - --cfg armv6m
+```
+
+Important are the variable RUSTC_TARGET and CARGO_TARGET_PREFIX.
+Add `--cfg foo` as needed.
+
+Chances are that if you need to add this, you'll also have to add support for the architecture to `riot-rs-threads`.
+
+## Adding an embassy architecture
+
+RIOT-rs currently supports nrf, stm32, rp and the externally (from embassy) maintained esp-rs.
```

Comment:
That sentence is a bit confusing. Do you mean embassy supports it?

Reply:
I tried to clarify this